src/Datadog.Trace/SpanContext.cs

```diff
@@ -22,11 +22,13 @@ public class SpanContext : ISpanContext
         /// <param name="spanId">The propagated span id.</param>
         /// <param name="samplingPriority">The propagated sampling priority.</param>
         /// <param name="serviceName">The service name to propagate to child spans.</param>
-        public SpanContext(ulong? traceId, ulong spanId, SamplingPriority? samplingPriority = null, string serviceName = null)
+        /// <param name="origin">The propagated origin of the trace.</param>
+        public SpanContext(ulong? traceId, ulong spanId, SamplingPriority? samplingPriority = null, string serviceName = null, string origin = null)
```

Comment:
Adding the new optional parameter `origin` is non-breaking if users always recompile, but it is a binary breaking change. Are we ok with this (admittedly small) risk? Or should we add a ctor overload?

Reply:
I don't have a strong opinion about it.

Reply:
The cost of adding another ctor is so small, that maybe worth not take the risk.

Reply:
Ok, I've restore the previous public ctor and introduce a new internal ctor for the origin header, that way we avoid the breaking change and we don't expose the setter of the origin property.